### ISSUE TITLE
Update SoLoader to v0.10.3 - /data/user/0/com.fatmaprn/lib-0/dso_lock: open failed: EACCES (Permission denied)

### ISSUE THREAD

**Issue by kwef** [facebook/react-native#32735]
### Description

This crash on app startup comes from SoLoader in React Native (and related libraries) where the SoLoader fail to load a library because it cannot be found. Specifically, the crash comes from [this line](https://github.com/facebook/SoLoader/commit/efdd618edc867ebbe7195eec0c76a1b3fe3684f2#diff-e724fcd379b9a7948c501e0f4b827c1d41a257833d35946b2fef2285b3d395c3L489), where it checks a dso-lock file and is unable to find it in some situations. It seemed to affect only newest Samsung models e.g. S21, and Google Pixel 6 or 6 Pro. which run on Android 11 or Android 12. This crash was noticed on Bugsnag after we updated to RN 0.66.2 and changed the targetSdkVersion to 30: 

```
java.lang.RuntimeException: Unable to create application com.fatmaprn.MainApplication: java.lang.RuntimeException: java.io.FileNotFoundException: /data/user/0/com.fatmaprn/lib-0/dso_lock: open failed: EACCES (Permission denied)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7434)
        at android.app.ActivityThread.access$1700(ActivityThread.java:309)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2268)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:226)
        at android.os.Looper.loop(Looper.java:313)
        at android.app.ActivityThread.main(ActivityThread.java:8582)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:563)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1133)

Caused by: java.lang.RuntimeException: java.io.FileNotFoundException: /data/user/0/com.fatmaprn/lib-0/dso_lock: open failed: EACCES (Permission denied)
        at com.facebook.soloader.SoLoader.init(SoLoader.java:219)
        at com.fatmaprn.MainApplicationBase.onCreate(MainApplicationBase.java:81)
        at com.fatmaprn.MainApplication.onCreate(MainApplication.java:10)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1211)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7429)
        at android.app.ActivityThread.access$1700(ActivityThread.java:309)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2268)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:226)
        at android.os.Looper.loop(Looper.java:313)
        at android.app.ActivityThread.main(ActivityThread.java:8582)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:563)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1133)

Caused by: java.io.FileNotFoundException: /data/user/0/com.fatmaprn/lib-0/dso_lock: open failed: EACCES (Permission denied)
        at libcore.io.IoBridge.open(IoBridge.java:575)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:236)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:186)
        at com.facebook.soloader.FileLocker.<init>(FileLocker.java:36)
        at com.facebook.soloader.FileLocker.lock(FileLocker.java:32)
        at com.facebook.soloader.UnpackingSoSource.prepare(UnpackingSoSource.java:430)
        at com.facebook.soloader.SoLoader.initSoSources(SoLoader.java:327)
        at com.facebook.soloader.SoLoader.init(SoLoader.java:207)
        at com.facebook.soloader.SoLoader.init(SoLoader.java:189)
        at com.facebook.soloader.SoLoader.init(SoLoader.java:217)
        at com.fatmaprn.MainApplicationBase.onCreate(MainApplicationBase.java:81)
        at com.fatmaprn.MainApplication.onCreate(MainApplication.java:10)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1211)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7429)
        at android.app.ActivityThread.access$1700(ActivityThread.java:309)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2268)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:226)
        at android.os.Looper.loop(Looper.java:313)
        at android.app.ActivityThread.main(ActivityThread.java:8582)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:563)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1133)
--
```

 The fix made in the Soloader library v0.10.3, and by forcing the update in `app/android/app/build.gradle` we were able to fix it
 
 ```
     implementation('com.facebook.soloader:soloader:0.10.3') {
        version {
            strictly '0.10.3'
        }
    }
 ```
Related Issues: https://github.com/facebook/SoLoader/issues/74
Fix in SoLoader: https://github.com/facebook/SoLoader/commit/efdd618edc867ebbe7195eec0c76a1b3fe3684f2#

Please update the [SoLoader](https://github.com/facebook/react-native/blob/1465c8f3874cdee8c325ab4a4916fda0b3e43bdb/ReactAndroid/gradle.properties#L16) to v0.10.3

### Version

0.66.2

### Output of `react-native info`

System:
    OS: macOS 11.1
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Memory: 29.29 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.11 - /usr/local/bin/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
    Watchman: 2021.10.18.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /Users/karinwefald/.gem/ruby/2.7.1/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    Android SDK:
      API Levels: 23, 25, 26, 27, 28, 29, 30
      Build Tools: 23.0.1, 23.0.2, 25.0.0, 25.0.1, 25.0.2, 25.0.3, 26.0.1, 26.0.2, 26.0.3, 27.0.1, 27.0.3, 28.0.3, 29.0.2, 29.0.3, 30.0.2
      System Images: android-23 | Intel x86 Atom_64, android-23 | Google APIs Intel x86 Atom_64, android-28 | Google Play Intel x86 Atom, android-29 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.6858069
    Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_152 - /Library/Java/JavaVirtualMachines/jdk1.8.0_152.jdk/Contents/Home/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2
    react-native: 0.66.2 => 0.66.2
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

It seemed to affect only newest Samsung models e.g. S21, and Google Pixel 6 or 6 Pro. which run on Android 11 or Android 12. This crash was noticed on Bugsnag after we updated to RN 0.66.2 and changed the targetSdkVersion to 30.

**Comment by chiralium** [comment]
Same problem

**Comment by cortinico** [comment]
> Same problem

This has been addressed in f45889ef95ec694520e91b0032e591a087e088e5
Will be released in RN 0.68.0

**Comment by shakawy** [comment]
@cortinico can't this be released in a sub version in RN 0.76? 

Because of the new architecture i can not upgrade to RN 0.68.0 now because im using third party packages that are using bridges to native and they weren't upgraded to the new architecture

**Comment by cortinico** [comment]
> Because of the new architecture i can not upgrade to RN 0.68.0

You can use RN 0.68 **without** the New Architecture. That's an opt-in for now 👍
